src/Uno.UI/UI/Xaml/Controls/ToolTip/ToolTip.cs

```diff
@@ -73,10 +129,600 @@ private void AttachToPopup()
+			intersectionRect.Height = maxY;
+
+			intersectionRect.Intersect(toolTipRect);
+			if ((Math.Abs(intersectionRect.Width - toolTipRect.Width) < TOOLTIP_TOLERANCE) &&
+				(Math.Abs(intersectionRect.Height - toolTipRect.Height) < TOOLTIP_TOLERANCE))
+			{
+				// The placement algorithm operates in LTR mode (with transformed data if it
+				// is really in RTL mode), so we also need to transform the X value it returns.
+				if (bIsRTL)
+				{
+					left = screenWidth - left;
+				}
+
+				// ToolTip is completely inside the plug-in
+				spPopup.VerticalOffset = top;
+				spPopup.HorizontalOffset = left;
+			}
+			else
+			{
+				if (top + toolTipRect.Height > maxY)
+				{
+					// If the lower edge of the plug-in obscures the ToolTip,
+					// it repositions itself to align with the upper edge of the bounding box of the mouse.
+					top = maxY - toolTipRect.Height - TOOLTIP_TOLERANCE;
+				}
+
+				if (top < 0)
+				{
+					// If the upper edge of Plug-in obscures the ToolTip,
+					// the control repositions itself to align with the upper edge.
+					// align with the top of the plug-in
+					top = 0;
+				}
+
+				if (left + toolTipRect.Width > maxX)
+				{
+					// If the right edge obscures the ToolTip,
+					// it opens in the opposite direction from the obscuring edge.
+					left = maxX - toolTipRect.Width - TOOLTIP_TOLERANCE;
+				}
+
+				if (left < 0)
+				{
+					// If the left edge obscures the ToolTip,
+					// it then aligns with the obscuring screen edge
+					left = 0;
+				}
+
+				// if right/bottom doesn't fit into the plug-in bounds, clip the ToolTip
+				{
+					var clipCalculationsRect = new Rect(left, top, toolTipRect.Width, toolTipRect.Height);
+					CalculateTooltipClip(clipCalculationsRect, maxX, maxY);
+				}
+
+				// The placement algorithm operates in LTR mode (with transformed data if it
+				// is really in RTL mode), so we also need to transform the X value it returns.
+
+				if (bIsRTL)
+				{
+					left = screenWidth - left;
+				}
+
+				// position the parent Popup
+				spPopup.VerticalOffset = top + verticalOffset;
+				spPopup.HorizontalOffset = left + horizontalOffset;
+			}
+		}
+
+		// Sets the location of the ToolTip's Popup within the Xaml window.
+		private void PerformNonMousePlacementWithPopup(
+			Rect? pTargetRect, Rect pDimentions, PlacementMode placement)
+		{
+
+			var tooltipActualWidth = pDimentions.Width;
+			var tooltipActualHeight = pDimentions.Height;
+			var horizontalOffset = pDimentions.Left;
+			var verticalOffset = pDimentions.Top;
+
+			var bIsRTL = false;
+
+			var origin = default(Point);
+			var rcDockTo = default(Rect);
+			var szFlyout = default(Size);
+
+			var spTarget = _owner as FrameworkElement;
+			_Debug.Assert(spTarget != null, "pTarget expected to be non-null in ToolTip_Partial::PerformPlacement()");
+			if (spTarget != null)
+			{
+				// UNO TODO
+				//	        xaml::FlowDirection targetFlowDirection = xaml::FlowDirection_LeftToRight;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5404644)